storage/database/db_manager.go

```diff
@@ -534,7 +534,16 @@ func (stdBatch *stateTrieDBBatch) Put(key []byte, value []byte) error {
 			errResult = err
 		}
 	}
+	return errResult
+}
 
+func (stdBatch *stateTrieDBBatch) Delete(key []byte) error {
+	var errResult error
+	for _, batch := range stdBatch.batches {
+		if err := batch.Delete(key); err != nil {
+			errResult = err
```

Comment:
How about returning an error early at this point?

Reply:
I just followed the structure of `stateTrieDBBatch.Put` for consistency.